src/main/scala/is/hail/sparkextras/ContextRDD.scala

```diff
@@ -0,0 +1,159 @@
+package is.hail.sparkextras
+
+import is.hail.io._
+import is.hail.expr.types._
+import is.hail.utils._
+
+import org.apache.spark._
+import org.apache.spark.rdd._
+import org.apache.spark.storage._
+import scala.reflect.ClassTag
+
+object ContextRDD {
+  def empty[C <: AutoCloseable, T: ClassTag](
+    sc: SparkContext,
+    mkc: () => C
+  ): ContextRDD[C, T] =
+    new ContextRDD(sc.emptyRDD[C => Iterator[T]], mkc)
+
+  // this one weird trick permits the caller to specify T without C
+  sealed trait Empty[T] {
+    def apply[C <: AutoCloseable](
+      sc: SparkContext,
+      mkc: () => C
```

Comment:
On last interface idea: Can you have something like `C: Constructible` with:

```
trait Constructible {
  def make: C
}
```

and

```
object RVDContext {
  implicit constructible: Constructible[RVDContext] = new Trait {
    def make: RVDContext = RVDContext()
  }
}
```

I've never fully internalized the implicit resolution rules but I think then things maybe just work and you don't need to be explicit about mkc anymore.

Reply:
Yeah this seems pretty reasonable.

Reply:
I called it `Pointed`